HeimdallrTests/Core/HeimdallrSpec.swift

```diff
@@ -725,6 +725,34 @@ class HeimdallrSpec: QuickSpec {
                         expect(result?.error?.code).to(equal(OAuthErrorInvalidClient))
                     }
                 }
+
+                context("when issueing multiple requests at the same time") {
+                    it("only issues one refresh token call") {
```

Comment:
You are absolutely right, but I couldn't figure out how to do that without massively intervening with the code under test. Hence the proxy that essentially just looks for the problem we tried to tackle.
Any ideas on how to actually test the sequential execution?